### DISCUSSION TITLE
`--dry-run` (or something similar) for `odo push`

### DISCUSSION THREAD

**Discussion by dharmit** [redhat-developer/odo#4842] [Ideas]
With #4819 going in, odo will finally be able to provide the workflow we had long been discussing:

* A user (say `user 1`) can add devfile.yaml with following information into their VCS (git):
  * component (nodejs, Spring Boot, etc.)
  * Operator backed service (database, queue, etc.)
  * link information (a `servicebinding` between the component and a service)
* Another user from their team (say `user 2`) could clone this repository from the VSC and have the component, the service and link ready to use by simply doing `odo create --devfile devfile.yaml && odo push`.

However, there's one caveat. If `user 2` doesn't have the Operator(s) used to create the service(s) in devfile, `odo push` will fail. If odo could somehow tell `user 2` what it will require to prevent `odo push` from failing, it would help user set things up before doing `odo push`.

`odo push --dry-run` would check the cluster for the availability of CRs mentioned in `components.kubernetes.inlined` and produce an output asking the user to install/setup a few things before doing `odo push`.

`--dry-run` is just an example. We could call the flag something else.

**Comment by kadel** [comment]
> `odo push --dry-run` would check the cluster for the availability of CRs mentioned in `components.kubernetes.inlined` and produce an output asking the user to install/setup a few things before doing `odo push`.

This check should be part of a normal `odo push` flow

**Reply by dharmit** [reply]
Yeah, it just occurred to me after talking about this on the call. We don't need an extra flag if `odo push` conveys it in a nice way.
